dags/synaptor_ops.py

```diff
@@ -21,13 +21,145 @@
 TASK_QUEUE_NAME = "synaptor"
 
 
-# Python callables (for PythonOperators)
-def generate_ngl_link_op() -> None:
+# Op functions
+def generate_ngl_link(
+    net_output_path: str,
+    seg_path: str,
+    workflowtype: str,
+    storagedir: str,
+    img_path: Optional[str] = None,
+    voxelres: Optional[tuple[int, int, int]] = None,
+) -> None:
     """Generates a neuroglancer link to view the results."""
-    pass
+    layers = [
+        ImageLayer("network output", net_output_path),
+        SegLayer("synaptor segmentation", seg_path),
+    ]
+
+    if img_path:
+        layers = [ImageLayer("image", img_path)] + layers
+
+    payload = generate_ng_payload(layers)
+
+    if "Assignment" in workflowtype:
+        presyn_pts, postsyn_pts = read_pts(storagedir)
+        payload = add_annotation_layer(payload, presyn_pts, postsyn_pts, voxelres)
+
+    basename = dump_payload(payload, storagedir)
+
+    slack_message(wrap_payload(os.path.join(storagedir, basename)), broadcast=True)
+
+
+def dump_payload(payload: dict, storagedir: str) -> str:
+    from cloudfiles import CloudFiles
+    basename = "ng.json"
+
+    cf = CloudFiles(storagedir)
```

Comment:
I think it is better to use `upload_json` from igneous_and_cloudvolume, it suppose to `mount` necessary secrets before upload files (after you populate the `mount_secrets` variables with the secrets you have).